File: LENTABP10/src/main/java/com/lenta/lentabp10/models/repositories/ITaskProductRepository.java
package com.lenta.lentabp10.models.repositories;

import com.lenta.shared.models.core.IProduct;
import com.lenta.shared.models.core.ProductInfo;

import java.util.List;

public interface ITaskProductRepository {
    List<ProductInfo> getProducts();
    ProductInfo findProduct(IProduct product);
    boolean addProduct(ProductInfo product);
    boolean deleteProduct(ProductInfo product);
    void clear();
    ProductInfo get(int index);
    int lenght();
}

File: writeoff_breakage/src/main/java/com/lenta/bp10/models/repositories/ITaskProductRepository.kt
package com.lenta.bp10.models.repositories

import com.lenta.shared.models.core.ProductInfo

interface ITaskProductRepository {
    fun getProducts(): List<ProductInfo>
    fun findProduct(product: ProductInfo): ProductInfo
    fun addProduct(product: ProductInfo): Boolean
    fun deleteProduct(product: ProductInfo): Boolean
    fun clear()
    operator fun get(index: Int): ProductInfo
    fun lenght(): Int
}